### PR TITLE
Alternative and Filterable to ReaderObservable added

### PR DESCRIPTION
**Pull request by mlegenhausen**

### CODE REVIEW

File: src/ReaderObservable.ts [hunk 1]
```diff
@@ -149,18 +153,56 @@ export function chainTaskK<A, B>(
...
 
-export const readerObservable: Monad2<URI> & MonadObservable2<URI> = {
+export const readerObservable: Monad2<URI> & Alternative2<URI> & Filterable2<URI> & MonadObservable2<URI> = {
   URI,
   map: T.map,
   of,
   ap: T.ap,
   chain: T.chain,
+  zero: () => R.observable.zero,
+  alt: (fx, f) => r => merge(fx(r), f()(r)),
```
Comment (gcanti) [inline src/ReaderObservable.ts]: I'd reuse the `R.observable` instance here

```ts
alt: (fx, f) => r => R.observable.alt(fx(r), () => f()(r))
```

File: src/ReaderObservable.ts [hunk 2]
```diff
@@ -149,18 +153,56 @@ export function chainTaskK<A, B>(
...
+    right: readerObservable.filterMap(fa, a =>
+      pipe(
+        f(a),
+        E.fold(() => O.none, O.some)
+      )
+    )
+  }),
+  partition: <R, A>(fa: ReaderObservable<R, A>, p: Predicate<A>) =>
+    readerObservable.partitionMap(fa, E.fromPredicate(p, identity)),
+  filterMap: <R, A, B>(fa: ReaderObservable<R, A>, f: (a: A) => O.Option<B>) => (r: R) =>
```
Comment (gcanti) [inline src/ReaderObservable.ts]: Nitpick: no need for type annotations here

```ts
filterMap: (fa, f) => r => R.observable.filterMap(fa(r), f)
```

**Review by gcanti — CHANGES_REQUESTED** [review]

File: docs/modules/Observable.ts.md [hunk 3]
```diff
@@ -13,28 +13,28 @@ Added in v0.6.0
...
-- [fromOption](#fromoption)
-- [fromTask](#fromtask)
-- [getMonoid](#getmonoid)
-- [map](#map)
-- [observable](#observable)
-- [of](#of)
-- [partition](#partition)
-- [partitionMap](#partitionmap)
-- [separate](#separate)
-- [toTask](#totask)
```
Comment (gcanti) [inline docs/modules/Observable.ts.md]: oh, looks like you have a old version of `docs-ts` installed, could you please install `docs-ts@latest` and generate the docs again?

**Review by gcanti — CHANGES_REQUESTED** [review]

File: docs/modules/Observable.ts.md (hunk 3 above)
Comment (gcanti) [reply docs/modules/Observable.ts.md]: current version in `package.json` is `0.3.4`

**Comment by mlegenhausen** [comment]
Looks better now. I also fixed a few wrong `since` comments.

**Review by gcanti — APPROVED** [review]

**Comment by gcanti** [comment]
Thanks @mlegenhausen, released.

**Pull request merged**